iothub/service/design.md

```diff
@@ -0,0 +1,119 @@
+
+## Service Client
+
+```csharp
+/// <summary>
+/// Create ServiceClient using AAD token credentials and the specified Transport Type.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential">AAD <see cref="TokenCredential"> to authenticate with IoT hub.</param>
+/// <param name="transportType">Specifies whether Amqp or Amqp_WebSocket_Only transport is used.</param>
+/// <param name="transportSettings">Specifies the AMQP and HTTP proxy settings for Service Client.</param>
+/// <param name="options">The options that allow configuration of the service client instance during initialization.</param>
+/// <returns>An instance of <see cref="ServiceClient"/>.</returns>
+public static ServiceClient Create(
+    string hostName,
+    TokenCredential credential,
+    TransportType transportType,
+    ServiceClientTransportSettings transportSettings = default,
+    ServiceClientOptions options = default)
+
+/// <summary>
+/// Create ServiceClient using SAS token credentials and the specified Transport Type.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential"><see cref="AzureSasCredential"> to authenticate with IoT hub.</param>
+/// <param name="transportType">Specifies whether Amqp or Amqp_WebSocket_Only transport is used.</param>
+/// <param name="transportSettings">Specifies the AMQP and HTTP proxy settings for Service Client.</param>
+/// <param name="options">The options that allow configuration of the service client instance during initialization.</param>
+/// <returns>An instance of <see cref="ServiceClient"/>.</returns>
+public static ServiceClient Create(
+    string hostName,
+    AzureSasCredential credential,
+    TransportType transportType,
+    ServiceClientTransportSettings transportSettings = default,
+    ServiceClientOptions options = default)
+```
+
+## Registry Manager
+
+```csharp
+/// <summary>
+/// Creates an instance of <see cref="RegistryManager"/>.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential">AAD <see cref="TokenCredential"> to authenticate with IoT hub.</param>
+/// <param name="transportSettings">The HTTP transport settings.</param>
+/// <returns>An instance of <see cref="RegistryManager"/>.</returns>
+public static RegistryManager Create(
+    string hostName,
+    TokenCredential credential,
+    HttpTransportSettings transportSettings)
+
+/// <summary>
+/// Creates an instance of <see cref="RegistryManager"/>.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential"><see cref="AzureSasCredential"> to authenticate with IoT hub.</param>
+/// <param name="transportSettings">The HTTP transport settings.</param>
+/// <returns>An instance of <see cref="RegistryManager"/>.</returns>
+public static RegistryManager Create(
+    string hostName,
+    AzureSasCredential credential,
+    HttpTransportSettings transportSettings)
```

Comment:
Is there a default HttpTransportSettings we could have here? For users who don't use proxies, it would simplify their code a bit

Reply:
Good point. I don't know so I'll find out.

Reply:
Yes there is, so made it optional. Thanks for calling this out.